lib/manager/bundler/locked-version.ts

```diff
@@ -0,0 +1,43 @@
+import { isVersion } from '../../versioning/ruby';
+
+const DEP_REGEX = '(?<=\\().*(?=\\))';
+export function extractLockFileEntries(
+  lockFileContent: string
+): Map<string, string> {
+  const gemLock = new Map<string, string>();
+  let parsingGemSection = false;
+  lockFileContent.split('\n').forEach(eachLine => {
+    const whitespace = eachLine.indexOf(eachLine.trim());
+
+    const isGemLine = eachLine.trim().startsWith('GEM');
+
+    if (parsingGemSection === false && whitespace === 0 && isGemLine) {
+      parsingGemSection = isGemLine;
+    }
+
+    if (parsingGemSection === true && whitespace === 0 && !isGemLine) {
+      parsingGemSection = false;
+    }
+
+    // as per original ruby lockfile parser,a line whitespace 2,4,6 contains dependencies.
+    if (whitespace === 4 && parsingGemSection) {
+      // checking if the dependency string has version or not
+      const depString = eachLine.match(DEP_REGEX);
```

Comment:
Wouldn't it be more performat to instanciate the regex on module or function scope and reuse it?

@rarkins @JamieMagee What do you think?

Reply:
This regex is extracting the version information from each string, either way, I believe the complexity is still O(n). I tried to keep its execution minimal, so I have placed this as nested as possible. Couldn't figure out a simpler approach than this